### PR TITLE
Explore: Query row using character refId instead of number

### LINKED ISSUE #15573: "explore" feature: change target refid to letters like in dashboards
### What Grafana version are you using?
6.0.0-beta3

### What datasource are you using?
custom datasource as described [here](http://docs.grafana.org/plugins/developing/datasources/#query)

### What did you do?
Right now I'm using the latest stable 5.4.3 build with a custom datasource. The datasource identifies the queries by their _targets refid's_ as described [here](http://docs.grafana.org/plugins/developing/datasources/#query):
```
{
  "range": { "from": "2015-12-22T03:06:13.851Z", "to": "2015-12-22T06:48:24.137Z" },
  "interval": "5s",
  "targets": [
    { "refId": "B", "target": "upper_75" },
    { "refId": "A", "target": "upper_90" }
  ],
  "format": "json",
  "maxDataPoints": 2495 //decided by the panel
}
```
While trying the new 6.0.0-beta3 I noticed that all dashboards still work, but the new "explore" feature doesn't. This is being caused by the different query behaviour: instead of using **letters** (A and B in the above example), the _target refid's_ are now incremented **numbers**:
```
...
    { "refId": 1, "target": "upper_75" },
    { "refId": 2, "target": "upper_90" }
...
```
Our custom datasource makes use of the _target refid's_ and **does not accept numbers** here.

### What was the expected result?
I expect the refid's to be named the same way, regardless of the use case ("explore" or dashboards/panels).

### What happened instead?
The refid's are numbers in the "explore" feature and letters in dashboards/panels.

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
Fixes an issue where query rows had a number as refId instead of a char.
**Which issue(s) this PR fixes**:
Fixes #15573 

**Special notes for your reviewer**:

**Release note**:
```release-note
none
```

### CODE REVIEW

File: public/app/core/utils/explore.ts [hunk 1]
```diff
@@ -225,22 +226,18 @@ export function generateKey(index = 0): string {
   return `Q-${Date.now()}-${Math.random()}-${index}`;
 }
 
-export function generateRefId(index = 0): string {
-  return `${index + 1}`;
-}
-
-export function generateEmptyQuery(index = 0): { refId: string; key: string } {
-  return { refId: generateRefId(index), key: generateKey(index) };
+export function generateEmptyQuery(queries: DataQuery[], index = 0): { refId: string; key: string } {
```
Comment (torkelo) [inline public/app/core/utils/explore.ts]: should this not return a DataQuery?

**Review by torkelo — APPROVED** [review]
Looks good! just the return type of generateEmptyQuery that looks strange.

File: public/app/core/utils/explore.ts (hunk 1 above)
Comment (peterholmberg) [reply public/app/core/utils/explore.ts]: Hm, did not consider the return type of this function but yes it could return a DataQuery. Pushed fix.

**Comment by davkal** [comment]
What happens on the 27th row? `getNextRefIdLetter` callees should handle the case when `undefinded` is returned or the ref generator should produce `AA` etc.

In general I'm regretting that I introduced `key` as part of an empty query, because it's a react-ism that should not leak into core modules.

File: public/app/core/utils/query.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import _ from 'lodash';
+import { DataQuery } from '@grafana/ui/';
+
+export const getNextRefIdLetter = (queries: DataQuery[]): string => {
```
Comment (davkal) [inline public/app/core/utils/query.ts]: Could use a test.

File: public/app/features/dashboard/state/PanelModel.ts [hunk 3]
```diff
@@ -5,6 +5,7 @@ import _ from 'lodash';
 import { Emitter } from 'app/core/utils/emitter';
 import { DataQuery, TimeSeries, Threshold, ScopedVars, PanelTypeChangedHook } from '@grafana/ui';
 import { TableData } from '@grafana/ui/src';
+import { getNextRefIdLetter } from '../../../core/utils/query';
```
Comment (davkal) [inline public/app/features/dashboard/state/PanelModel.ts]: Should probably be `'app/core/utils/query'`

**Comment by ryantxu** [comment]
> What happens on the 27th row? `getNextRefIdLetter` callees should handle the case when `undefinded` is returned or the ref generator should produce `AA` etc.
> 

Check: https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/Table/Table.story.tsx#L22

That aimed to copy the excel column naming strategy -- likely should be in a common utility function :)

**Comment by torkelo** [comment]
@ryantxu we cannot use that approach as query refId cannot change as other queries can refer to them, we cannot simply take one based on index. Need to check and take first one that is not already in use.

**Pull request merged**